### PR TITLE
Fix Javadoc Issues

### PR DESCRIPTION
**Pull request by ImalshaG**
This PR adds the javadoc plugin and fixes javadoc related issues. The existing SCR doc comments are changed to OSGI component annotation.

Related Issue: https://github.com/wso2/product-is/issues/14203

### CODE REVIEW

File: components/org.wso2.carbon.identity.provisioning.connector.scim2/src/main/java/org/wso2/carbon/identity/provisioning/connector/scim2/internal/SCIM2ConnectorServiceComponent.java [hunk 1]
```diff
@@ -25,9 +25,7 @@
 import org.wso2.carbon.identity.provisioning.connector.scim2.SCIM2ProvisioningConnectorFactory;
 
 /**
- * @scr.component name=
- * "org.wso2.carbon.identity.provisioning.connector.scim2.internal.SCIM2ConnectorServiceComponent"
- * immediate="true"
+ * OSGI Service Component that handles SCIM2 Connector activation.
```
Comment (piraveena) [inline components/org.wso2.carbon.identity.provisioning.connector.scim2/src/main/java/org/wso2/carbon/identity/provisioning/connector/scim2/internal/SCIM2ConnectorServiceComponent.java:28]: is the service component activates without the scr annotation? we need to change the scr annotation to the latest one

File: pom.xml [hunk 2]
```diff
@@ -113,6 +113,12 @@
                 <artifactId>okio</artifactId>
                 <version>${okio.version}</version>
             </dependency>
+            <dependency>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>org.apache.felix.scr.ds-annotations</artifactId>
+                <version>${apache.felix.scr.ds.annotations.version}</version>
+            </dependency>
+
```
Comment (piraveena) [inline pom.xml:121]: Remove new line

File: pom.xml [hunk 3]
```diff
@@ -255,6 +276,7 @@
         <powermock.version>1.6.5</powermock.version>
         <okio.version>1.9.0</okio.version>
         <json.wso2.version.range>[3.0.0.wso2v1, 4.0.0)</json.wso2.version.range>
+        <apache.felix.scr.ds.annotations.version>1.2.4</apache.felix.scr.ds.annotations.version>
```
Comment (piraveena) [inline pom.xml:279]: Is this the latest version?

**Review by piraveena — APPROVED** [review]

File: pom.xml (hunk 3 above)
Comment (ImalshaG) [reply pom.xml]: Bumped to the latest: 1.2.10

**Pull request merged**